### PR TITLE
HTTP client for the ISCN registration server

### PR DESCRIPTION
**Pull request by anaulin**
This PR addresses https://github.com/starlinglab/starling-integrity-api/issues/53 by providing a library that wraps the call to an HTTP server that acts as a proxy for ISCN registration.

The server is expected to be an instance of https://github.com/likecoin/iscn-js/tree/master/sample/server which holds the wallet that will "sign" the registration. To run this server, after doing `npm install`, invoke: `$ MNEMONIC_SEED_WORD=<your wallet mnemonic> node src/index.js`.

The `registration` dictionary itself will have a schema like the one discussed in https://github.com/starlinglab/starling-integrity-api/issues/53#issuecomment-1058107379 . We will implement assembling this registration object once we are ready to wire the yet-to-be-written "batcher" code to the ISCN registration.

With this change and a locally running ISCN registration server, it is possible to do a registration with:
```
# Enter our Python environment
$ pipenv shell
$ python
>>> from starlingcaptureapi.iscn import Iscn
>>> Iscn.register(<your registration dict>)
```

An example registration looks something like (see https://github.com/starlinglab/starling-integrity-api/issues/53#issuecomment-1058107379 for a detailed discussion of what we intend this to look like, although for testing you can do something simpler):
```
{
    "recordNotes": "Initial registration.",
    "contentFingerprints": [
        "ipfs://<SOME_CID>"
    ],
    "contentMetadata": {
        "@context": "https://schema.org",
        "@type": "CreativeWork",
        "name": "<SOME NAME STRING>",
        "description": "<SOME DESCRIPTION>",
        "author": "https://starlinglab.org",
        "keywords": "foo, bar"
        "conditionsOfAccess": "Encrypted with AES-128."
    }
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ The server is configured entirely via environment variables. See [config.py](./s
 Most importantly, you will need to provide:
 * `CLAIM_TOOL_PATH`: A path to a fully working `claim_tool` binary. The server should have permissions to execute it, and it should be correctly configured with its keys.
 * `IMAGES_DIR`: A path to a directory to store images. The server will need write access to this directory. This will be the persistent storage for the received images with their attestations.
+* `ISCN_SERVER`: The instance of the ISCN server to send registration requests to. Typically, this will be `http://localhost:3000` if you are using the sample server at https://github.com/likecoin/iscn-js/tree/master/sample/server in its default configuration.
```
Comment (benhylau) [inline README.md:26]: @YurkoWasHere @anaulin will this be incorporated into deploy ansible?

File: starlingcaptureapi/iscn.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        Args:
+            registration: the complete contents of the ISCN registration; must comply with the
+                ISCN schema (https://github.com/likecoin/iscn-specs/tree/master/schema)
+
+        Returns:
+            True if the registration succeeded; False otherwise
+        """
+        response = requests.post(_REGISTER, json={"metadata": registration})
+
+        if not response.ok:
```
Comment (benhylau) [inline starlingcaptureapi/iscn.py:26]: I wonder how we may log failures in a way that can support a retry policy in the future. Same goes for Numbers and Open Timestamp registrations.

Comment (benhylau) [inline starlingcaptureapi/iscn.py:26]: I wonder how we may log failures in a way that can support a retry policy in the future. Same goes for Numbers and Open Timestamp registrations.

**Comment by benhylau** [comment]
In chat, William mentioned needing to keep an incrementing counter that needs to be unique per registration. I imagine that handled on the JS side? Do you know how it is handled?

**Comment by benhylau** [comment]
In chat, William mentioned needing to keep an incrementing counter that needs to be unique per registration. I imagine that handled on the JS side? Do you know how it is handled?

**Review by benhylau — APPROVED** [review]

**Comment by anaulin** [comment]
> In chat, William mentioned needing to keep an incrementing counter that needs to be unique per registration. I imagine that handled on the JS side? Do you know how it is handled?

My understanding is that, if handled, it would be on the JS side. I am not clear if it is actually handled. There is this comment in the code: https://github.com/likecoin/iscn-js/blob/master/sample/server/src/route.js#L61-L64

**Pull request merged**